packages/attestation-service/src/requestHandlers/get_attestation.ts

```diff
@@ -36,10 +36,47 @@ class GetAttestationRequestHandler {
     this.key = getAttestationKey(getRequest)
   }
 
-  async getAttestationRecord() {
-    return findAttestationByKey(this.key, {
+  async withAttestationAndSecurityCodeChecked(f: (attestation: AttestationModel) => any) {
```

Comment:
Optional Nit: This seems like a confusing interface where success means a void return, but an error throws with an error payload response?